scripts/production/Verify_Compara_REST_Endpoints.pl

```diff
@@ -263,25 +264,34 @@ sub fetch_leaf_hash_from_json {
     return $input_json;
 }
 
-sub verify_xml_leaf {
+
+# Fetch a species node using the production name node from a phyloxml input
+# if the searched species is not present it returns undef
+# so the presence of species can be tested by defined.
+sub fetch_species_node {
     my ($this_node, $species_name) = @_;
     my $nodes = ref($this_node) eq 'ARRAY' ? $this_node : [$this_node];
-
     foreach my $node ( @$nodes ) {
         if ( exists $node->{property} ) {
-            return 1 if ($node->{property}->{content} // '') eq $species_name;
+            return $node if (($node->{property}->{content} // '') eq $species_name);
         }
-
         if ( exists $node->{clade} ) {
-            my $verify_recursive = verify_xml_leaf($node->{clade}, $species_name);
-            return 1 if $verify_recursive;
-        } else {
-            foreach my $value ( values %$node ) {
-                return 1 if ($value->{property}->{content} // '') eq $species_name;
+            my $species_node = fetch_species_node($node->{clade}, $species_name);
+            return $species_node if defined $species_node;
+        }
+        else{# in case we are on an ancestral or species node
```

Comment:
For example, here it would have been clearer to leave the line with the `else` as it is, and add a comment on the next line
```suggestion
        } else {
            # in case we are on an ancestral or species node
```